e2e-polybft/e2e/bridge_test.go

```diff
@@ -1442,3 +1442,90 @@ func TestE2E_Bridge_NonMintableERC20Token_WithPremine(t *testing.T) {
 		}
 	})
 }
+
+func TestE2E_Bridge_NonNative(t *testing.T) {
+	const (
+		transfersCount       = 5
+		amount               = 100
+		epochSize            = 5
+		numberOfAttempts     = 4
+		stateSyncedLogsCount = 2
+	)
+
+	var (
+		stakeAmount = ethgo.Ether(500)
+	)
+
+	minter, err := wallet.GenerateKey()
+	require.NoError(t, err)
+
+	cluster := framework.NewTestCluster(t, 5,
+		framework.WithNumBlockConfirmations(0),
+		framework.WithEpochSize(epochSize),
+		framework.WithBridge(),
+		framework.WithBladeAdmin(minter.Address().String()),
+		framework.WithSecretsCallback(func(addrs []types.Address, tcc *framework.TestClusterConfig) {
+			for i := 0; i < len(addrs); i++ {
+				tcc.StakeAmounts = append(tcc.StakeAmounts, stakeAmount)
+			}
+		}),
+		framework.WithPredeploy(),
+	)
+	defer cluster.Stop()
+
+	cluster.WaitForReady(t)
+
+	firstValidator := cluster.Servers[0]
+
+	validatorAcc, err := validatorHelper.GetAccountFromDir(firstValidator.DataDir())
+	require.NoError(t, err)
+
+	relayer, err := txrelayer.NewTxRelayer(txrelayer.WithIPAddress(firstValidator.JSONRPCAddr()))
+	require.NoError(t, err)
+
+	mintFn := &contractsapi.MintRootERC20Fn{
+		To:     validatorAcc.Address(),
+		Amount: big.NewInt(100000),
+	}
+
+	mintInput, err := mintFn.EncodeAbi()
+	require.NoError(t, err)
+
+	nonNativeErc20 := ethgo.Address(contracts.ERC20Contract)
+
+	receipt, err := relayer.SendTransaction(
+		&ethgo.Transaction{
+			To:    &nonNativeErc20,
+			Input: mintInput,
+			Type:  ethgo.TransactionDynamicFee,
+		}, minter)
+	require.NoError(t, err)
+	require.Equal(t, uint64(types.ReceiptSuccess), receipt.Status)
+
+	polybftCfg, err := polybft.LoadPolyBFTConfig(path.Join(cluster.Config.TmpDir, chainConfigFileName))
+	require.NoError(t, err)
+
+	firstValidatorInfo, err := validatorHelper.GetValidatorInfo(validatorAcc.Ecdsa.Address(), relayer)
+	require.NoError(t, err)
+	require.True(t, firstValidatorInfo.Stake.Cmp(stakeAmount) == 0)
+
+	require.NoError(t, cluster.WaitForBlock(epochSize*1, 1*time.Minute))
+
+	require.NoError(t, firstValidator.Stake(polybftCfg, big.NewInt(1000), contracts.ERC20Contract))
```

Comment:
```suggestion
	require.NoError(t, firstValidator.Stake(polybftCfg, ethgo.Gwei(1), contracts.ERC20Contract))
```

also extract stake amount that you are staking to a variable. Not sure can we reuse `stakeAmount`?